### PR TITLE
Updated integrations/scm_repo_link structs and urls

### LINKED ISSUE #125: Update integrations/scm_repo_link structs and urls
Update theses modifications to keep a functional coherence:
- `Auth Service` has a new updated route for `integrations`: `/integrations` -> `/scm_integrations`
- `API` has renamed a field in `scm_repo_link`: `auth_integration_id` -> `auth_integration_uuid`

Add a missed method for the `SCMType` constant:
```go
func (t SCMType) Str() string {
	return string(t)
}
```
And rename all `Scm...` to `SCM...`

### PR DESCRIPTION
**Pull request by brandon-welsch**
Close #125

### CODE REVIEW

File: scm_integrations.go [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+type SCMIntegrationsService interface {
+	SCMIntegrationsList() ([]SCMIntegration, error)
+	SCMIntegrationsCreate(scmType SCMType, url string, accessToken string) (*SCMIntegration, error)
+	SCMIntegrationsDestroy(id string) error
+}
+
+var _ SCMIntegrationsService = (*Client)(nil)
+
+type SCMIntegration struct {
+	ID          string  `json:"id,omitempty"`
```
Comment (EtienneM) [inline scm_integrations.go]:
```suggestion
	ID          string  `json:"id"`
```

Are you sure about the `omitempty`? An ID seems mandatory to me

File: scm_integrations.go [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+	SCMType     SCMType `json:"scm_type"`
+	Url         string  `json:"url"`
```
Comment (EtienneM) [inline scm_integrations.go]:
```suggestion
	URL         string  `json:"url,omitempty"`
```
Go lint!!

Moreover, I think that `URL` and `AccessToken` are empty for GitHub.com and GitLab.com. SHouldn't they be `omitempty`?

File: scm_integrations.go [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+	AccessToken string  `json:"access_token"`
+	Uid         string  `json:"uid,omitempty"`
```
Comment (EtienneM) [inline scm_integrations.go]: Isn't it a UUID?

File: scm_integrations.go [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+	Username    string  `json:"username,omitempty"`
+	Email       string  `json:"email,omitempty"`
+	AvatarUrl   string  `json:"avatar_url,omitempty"`
```
Comment (EtienneM) [inline scm_integrations.go]:
```suggestion
	AvatarURL   string  `json:"avatar_url,omitempty"`
```

File: scm_integrations.go [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+	ProfileUrl  string  `json:"profile_url,omitempty"`
```
Comment (EtienneM) [inline scm_integrations.go]:
```suggestion
	ProfileURL  string  `json:"profile_url,omitempty"`
```

File: scm_integrations.go [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+type SCMIntegrationsRes struct {
+	SCMIntegrations []SCMIntegration `json:"scm_integrations"`
+}
+
+func (c *Client) SCMIntegrationsList() ([]SCMIntegration, error) {
+	var res SCMIntegrationsRes
+
+	err := c.AuthAPI().ResourceList("scm_integrations", nil, &res)
+	if err != nil {
+		return nil, errgo.Mask(err)
```
Comment (EtienneM) [inline scm_integrations.go]: Don' user `errgo.Mask`.

File: scm_integrations.go (hunk 1 above)
Comment (brandon-welsch) [reply scm_integrations.go]: The ID is automatically generated by the micro-service who manage it.

This struct is only used in the return of theses methods: list, show and create.

In create method, `scm_type` and `access_token` is mandatory to create an integration.

File: scm_integrations.go (hunk 2 above)
Comment (brandon-welsch) [reply scm_integrations.go]: `GitLab.com` and `GitHub.com` must not be created manually, but via an OAuth dance actually.

File: scm_integrations.go (hunk 3 above)
Comment (brandon-welsch) [reply scm_integrations.go]: `Uid` is the user id in the SCM integration.

For GitHub for example, `Uid` is the User ID on GitHub, and usually, it's an int.

File: scm_integrations.go [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+	SCMIntegrationsImportKeys(id string) ([]Key, error)
+}
+
+var _ SCMIntegrationsService = (*Client)(nil)
+
+type SCMIntegration struct {
+	ID          string  `json:"id"`
+	SCMType     SCMType `json:"scm_type"`
+	URL         string  `json:"url,omitempty"`
+	AccessToken string  `json:"access_token"`
```
Comment (EtienneM) [inline scm_integrations.go]: If you accepted the omitempty, on URL, shouldn't you add it here also?...

Comment (brandon-welsch) [reply scm_integrations.go]: The URL may well be empty, but access_token is still required

**Pull request merged**